manifests/server/config.pp

```diff
@@ -232,18 +232,25 @@
   }
 
   if $::puppet::server::git_repo {
-    # need to chown the $vardir before puppet does it, or else
-    # we can't write puppet.git/ on the first run
-
     include ::git
 
+    if $::puppet::server::manage_user {
+      Class['git'] -> User[$::puppet::server::user]
+    }
+
+    file { $::puppet::vardir:
+      ensure => directory,
+      owner  => 'root',
```

Comment:
Note I am changing this. In the old code it was owned by puppet:puppet but now it's root:root. This matches with the puppet-agent package (at least 5.5.10).